packages/roosterjs-editor-dom/lib/utils/normalizeTableSelection.ts

```diff
@@ -0,0 +1,31 @@
+import { Coordinates, TableSelection } from 'roosterjs-editor-types';
+
+/**
+ * Make the first Cell of a table selection always be on top of the last cell.
+ * @param input Table selection
+ * @returns Table Selection where the first cell is always going to be first selected in the table
+ * and the last cell always going to be last selected in the table.
+ */
+export default function normalizeTableSelection(
+    input: TableSelection,
+    table: HTMLTableElement
+): TableSelection {
+    const { firstCell, lastCell } = input;
+
+    let newFirst = {
+        x: Math.min(firstCell.x, lastCell.x),
+        y: Math.min(firstCell.y, lastCell.y),
+    };
+    let newLast = {
+        x: Math.max(firstCell.x, lastCell.x),
+        y: Math.max(firstCell.y, lastCell.y),
+    };
+
+    const checkIfExists = (coord: Coordinates) => table?.rows?.item(coord.y)?.cells.item(coord.x);
+
+    if (!checkIfExists(newFirst) || !checkIfExists(newLast)) {
+        throw new Error('Table selection provided is not valid');
```

Comment:
We can just ignore it here, no need to throw

Reply:
Just make sure the return value fall into the table is good enough.

Need to consider if the range is totally outside the table, what we should do?

Reply:
Sounds good, will do that